core/management/commands/backfill_commits.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+from redis import Redis
+from shared.config import get_config
+
+from core.models import Commit
+from services.task import TaskService
+
+log = logging.getLogger(__name__)
+
+
+def get_storage_redis():
+    return Redis.from_url(get_config("services", "redis_url"))
+
+
+def get_celery_redis():
+    return Redis.from_url(get_config("serivces", "celery_broker"))
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument("--batch-size", type=int)
+
+    def handle(self, *args, **options):
+        batch_size = options.get("batch_size", 1000)
+
+        storage_redis = get_storage_redis()
+        celery_redis = get_celery_redis()
+
+        queue_name = "backfill_commits"
+        queue_length = celery_redis.llen(queue_name)
+
+        if queue_length > 0:
+            log.warning(
+                "Backfill commits queue not drained",
+                extra=dict(
+                    queue_length=queue_length,
+                ),
+            )
+            return
+
+        commits = Commit.objects.all()
```

Comment:
Will this `all()` actually fetch ALL commits in the database? seems excessive, and commits are fairly complex objects...

I know you can define the query and the filters without actually fetching any data (and only do that at the end). I think it's worth doing that, if we're not doing so yet.

Reply:
This just returns a `QuerySet` which is a lazy representation of the actual SQL to be executed.  Below we do a `queryset[:batch_size]` which triggers the actual execution of the query and adds a `limit {batch_size}` to the SQL.

Reply:
https://docs.djangoproject.com/en/4.2/topics/db/queries/#limiting-querysets

Reply:
ah perfect, perfect. Thanks for the docs too